bundles/org.openhab.core.automation.rest/src/main/java/org/openhab/core/automation/rest/internal/ModuleTypeResource.java

```diff
@@ -62,41 +62,24 @@
  */
 @Path("module-types")
 @Api("module-types")
-@Component
-public class ModuleTypeResource implements RESTResource {
-
-    private ModuleTypeRegistry moduleTypeRegistry;
-    private LocaleService localeService;
-
-    @Context
-    private UriInfo uriInfo;
-
-    @Reference(cardinality = ReferenceCardinality.OPTIONAL, policy = ReferencePolicy.DYNAMIC)
-    protected void setModuleTypeRegistry(ModuleTypeRegistry moduleTypeRegistry) {
-        this.moduleTypeRegistry = moduleTypeRegistry;
-    }
-
-    protected void unsetModuleTypeRegistry(ModuleTypeRegistry moduleTypeRegistry) {
-        this.moduleTypeRegistry = null;
-    }
-
-    @Reference(cardinality = ReferenceCardinality.OPTIONAL, policy = ReferencePolicy.DYNAMIC)
-    protected void setLocaleService(LocaleService localeService) {
-        this.localeService = localeService;
-    }
-
-    protected void unsetLocaleService(LocaleService localeService) {
-        this.localeService = null;
-    }
+@Component(immediate = true)
+@JaxrsApplicationSelect("(osgi.jaxrs.name=" + RESTService.REST_APP_NAME + ")")
```

Comment:
For here and other places: Shouldn't something similar to

```java
@JaxrsApplicationSelect
@JaxrsName(RESTService.REST_APP_NAME)
```

work too?

My understanding has been that the property type has been introduced to not use constructs like `"(osgi.jaxrs.name=" + RESTService.REST_APP_NAME + ")"` anymore.

Reply:
Yes I guess so. Eclipse just went nuts on me and didn't resolve `JaxrsName` when I made those changes.